pkg/client/client.go

```diff
@@ -111,6 +129,26 @@ func (c *tlsBootstrapClientImpl) setupClientConnection(ctx context.Context) (*gr
 	return conn, nil
 }
 
+var newGetInstanceData = func(ctx context.Context, c *tlsBootstrapClientImpl, imdsURL string) (*datamodel.VMSSInstanceData, error) {
+	return c.imdsClient.GetInstanceData(ctx, baseImdsURL)
+}
+
+var newPbClientGetNonce = func(ctx context.Context, pbClient pb.AKSBootstrapTokenRequestClient, nonceRequest *pb.NonceRequest) (*pb.NonceResponse, error) {
+	return pbClient.GetNonce(ctx, nonceRequest)
```

Comment:
could theoretically make our own mockPBClient implementation for testing instead of having to do this